### PR TITLE
Remove "Controls not configured..." warning

### PR DESCRIPTION
**Pull request by MichaelArestad**
## What I did

I removed the "Controls not configured..." warning as it requires a developer to resolve and isn't always necessary to resolve. Instead, I would recommend an alternate approach via the CLI or maybe even something in the console.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [x] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: docs/snippets/common/button-story-hide-nocontrols-warning.js.mdx [hunk 1]
```diff
@@ -12,9 +12,5 @@ export default {
...
 };
 
-export const Large = {
-  parameters: {
-    controls: { hideNoControlsWarning: true },
-  },
-};
-```
\ No newline at end of file
+export const Large = {};
```
Comment (domyen) [inline docs/snippets/common/button-story-hide-nocontrols-warning.js.mdx:15]: The file seems to suggest that it's all about hiding the controls warning. Is this file still needed or perhaps we need to change the file name?

File: code/lib/blocks/src/components/ArgsTable/ArgsTable.stories.tsx [hunk 2]
```diff
@@ -61,7 +60,6 @@ export const InAddonPanel = {
 export const InAddonPanelWithWarning = {
```
Comment (domyen) [inline code/lib/blocks/src/components/ArgsTable/ArgsTable.stories.tsx:60]: This story name suggests a warning should be there, if we're removing it altogether we should remove the story as well

**Review by domyen — CHANGES_REQUESTED** [review]

File: code/lib/blocks/src/components/ArgsTable/ArgsTable.stories.tsx (hunk 2 above)
Comment (MichaelArestad) [reply code/lib/blocks/src/components/ArgsTable/ArgsTable.stories.tsx]: Good catch on both of these. Removed!

**Review by domyen — APPROVED** [review]

**Comment by domyen** [comment]
lgtm!

**Review by shilman — COMMENTED** [review]
@MichaelArestad This UI was added because when we released controls, it was been common for people to use CSF1 stories and wonder why their controls aren't showing up.

The situation may have changed as people have moved over to CSF2/3, which uses args and are therefore controllable. However, I know there are still plenty of CSF1 stories out there.

Putting this in the CLI won't work. Aside from technical hurdles, Using CSF1 is not a problem in itself, and using the controls addon for some of your stories and not others is also allowed.

Putting this in the browser console log is feasible. This message is intended for the specific case where you are looking for controls on your CSF1 story, don't see them, and are scratching your head why. Having users click to the devtools is a little indirect, but maybe it's fine.

Alternatively, perhaps we could improve the UI for this use case? As a user, when you click on the controls panel and there are no controls, it seems like an error, right? So putting something in the UI in this case (perhaps not an eyebrow) makes sense to me.

WDYT?

**Comment by MichaelArestad** [comment]
> Putting this in the browser console log is feasible. This message is intended for the specific case where you are looking for controls on your CSF1 story, don't see them, and are scratching your head why. Having users click to the devtools is a little indirect, but maybe it's fine.

This seems like a good compromise. The thing about this message is that it is visible all the time whether folks want to set up controls or not. It makes it feel a bit broken to use as it isn't something a viewer can or will address. I would much rather find a way to specifically help a developer understand why Controls isn't working or how to better set it up. The console isn't ideal because, as you mentioned, it's a bit hidden.

**Comment by shilman** [comment]
@MichaelArestad Something like this:

https://github.com/storybookjs/storybook/blob/next/code/addons/a11y/register.js#L4

**Comment by ndelangen** [comment]
I'm actually not a fan of putting anything in the console. 🙊

**Comment by MichaelArestad** [comment]
> I'm actually not a fan of putting anything in the console. 🙊

Out of curiosity, why? I also have another idea for where to put this if the console doesn't work.

**Comment by ndelangen** [comment]
It's extremely common to use storybook and `console.log` for debugging components.
If the storybook user does this, then us logging stuff to the console gets intermixed with what the user is doing.

I guess the user could filter the console noise, but I'd like to not make a habit out of dumping warnings for anything the user might need to be aware of in the console.

What is printed in the console has very little context, you need to explain that context, making the message even longer, making it harder to skip, increasing the noise, making everything less useful overall.

I fear that the "console spam" will give users the sensation that storybook is bloated, unstable..

"Did you forget to add argTypes?"
"Did you know X?"
"You should migrate feature Z"
"There is a new version available for Storybook"

This is not what the console is for.

It should not be our feedback / message / notification mechanism to the user.

So 2 main points:
- console has no context
- messages are often considered noise

**Comment by ndelangen** [comment]
Hey @MichaelArestad seems we've not really reached a end-state on this PR, but keeping it open for longer also won't help?

Please re-open if you feel this can move forward in some way still. ❤️

**Comment by MichaelArestad** [comment]
@ndelangen I have been working on this off and on over the last two weeks. I will probably create an issue, but I want to keep this open just a bit longer. Thanks!
